packages/tests/server/react/getQueryKey.test.tsx

```diff
@@ -39,6 +63,312 @@ const ctx = konn()
   .done();
 
 describe('getQueryKeys', () => {
+  test('no input', async () => {
+    const { proxy, App } = ctx;
+
+    function MyComponent() {
+      const happy1 = getQueryKey(proxy.post.all, undefined, 'query');
+      const happy2 = getQueryKey(proxy.post.all);
+
+      // @ts-expect-error - post.all has no input
+      const sad = getQueryKey(proxy.post.all, 'foo');
+
+      return (
+        <>
+          <pre data-testid="qKey1">{JSON.stringify(happy1)}</pre>
+          <pre data-testid="qKey2">{JSON.stringify(happy2)}</pre>
+        </>
+      );
+    }
+
+    const utils = render(
+      <App>
+        <MyComponent />
+      </App>,
+    );
+
+    await waitFor(() => {
+      expect(utils.getByTestId('qKey1')).toHaveTextContent(
+        JSON.stringify([['post', 'all'], { type: 'query' }]),
+      );
+      expect(utils.getByTestId('qKey2')).toHaveTextContent(
+        JSON.stringify([['post', 'all']]),
+      );
+    });
+  });
+
+  test('with input', async () => {
+    const { proxy, App } = ctx;
+
+    function MyComponent() {
+      const happy1 = getQueryKey(proxy.post.byId, { id: 1 }, 'query');
+
+      // doesn't really make sense but should still work
+      const happyIsh = getQueryKey(proxy.post.byId, { id: 1 });
+
+      // @ts-expect-error - post.byId has required input
```

Comment:
Did we want to allow fuzzy matching and make everything a deep partial and in that case should undefined also be allowed?

Reply:
im not familiar with this util in general

how does it currently behave if you fuzzy match?

Reply:
@TkDodo 

What is the expectation if we fuzzy match input `getQueryKey`? Is this even valid?

apologies, I am not well versed with this utility.

Reply:
One use case I'm thinking of is if you want to set some defaults for a procedure without caring about the specific input:

```ts
import { useQueryClient } from "@tanstack/react-query";
import { getQueryKey } from "@trpc/react-query";
import { trpc } from "~/utils/trpc";

function MyApp() {
  const qKey = getQueryKey(trpc.post.byId); // [["post", "byId"]], don't care about the input
  useQueryClient().setQueryDefaults(qKey, {
    // some opts
  });
}
```

Reply:
yeah this makes sense ⬆️